### PR TITLE
Issue#28 proactive conversion

### PR DESCRIPTION
**Pull request by tomdickman**

### CODE REVIEW

File: classes/conversion.php [hunk 1]
```diff
@@ -682,14 +695,41 @@ public function update_pending_conversions() : array {
     }
 
     /**
-     * Create the conversions.
-     * (Placeholder)
+     * Get the pathnamehases for files that have metadata extracted,
```
Comment (tomdickman) [inline classes/conversion.php]: spelling 'pathnamehases'

File: classes/conversion.php [hunk 2]
```diff
@@ -159,25 +159,34 @@ private function create_conversion(\stored_file $file) : void {
...
-        $convrecord->detect_phrases = $this->config->detectphrases;
-        $convrecord->detect_entities = $this->config->detectentities;
-        $convrecord->timecreated = $now;
-        $convrecord->timemodified = $now;
+        $cnvrec = new \stdClass();
+        $cnvrec->pathnamehash = $file->get_pathnamehash();
+        $cnvrec->contenthash = $file->get_contenthash();
+        $cnvrec->status = $this::CONVERSION_ACCEPTED;
+        $cnvrec->transcoder_status = $this::CONVERSION_ACCEPTED;
+        $cnvrec->transcribe_status =
```
Comment (tomdickman) [inline classes/conversion.php]: Could this be condensed to something like a mapping array: 
```php
$settingidconversionfields = ['transcribe_status' => 'transcribe', 'rekog_label_status' => 'detectlabels'...]
``` 
and a loop of:
```php
for each ($settingidconversionfields as $settingid => $conversionfield) {
    $cnvrec->$conversionfield = 
        $this->config->$settingid == 1 ? $this::CONVERSION_ACCEPTED : $this::CONVERSION_NOT_FOUND
}
```
Or is that overkill?

Comment (mattporritt) [reply classes/conversion.php]: good idea

**Comment by mattporritt** [comment]
code updated and pushed

**Pull request merged**